pg/src/test/java/org/bouncycastle/openpgp/test/PGPAeadTest.java

```diff
@@ -0,0 +1,481 @@
+                        System.out.println("Unexpected trailing packet.");
+                        System.out.println(o);
+                    }
+                }
+                catch (PGPException e)
+                {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    public static void printHex(byte[] bytes) {
+        boolean separate = true;
+        boolean prefix = true;
+        String hex = Hex.toHexString(bytes);
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < hex.length() / 2; i++) {
+            if (prefix && i % 8 == 0) {
+                sb.append("0x").append(String.format("%04X", i & 0xFFFFF)).append("   ");
+            }
+            sb.append(hex.substring(i * 2, i * 2 + 2));
+            if (separate) {
+                if ((i + 1) % 8 == 0) {
+                    sb.append('\n');
+                } else {
+                    sb.append(' ');
+                }
+            }
+        }
+        System.out.println(sb);
+    }
+
+    private static String algNames(int aeadAlg, int symAlg) {
+        String out = "";
+        switch (aeadAlg) {
+            case AEADAlgorithmTags.EAX:
+                out = "EAX";
+                break;
+            case AEADAlgorithmTags.OCB:
+                out = "OCB";
+                break;
+            case AEADAlgorithmTags.GCM:
+                out = "GCM";
+                break;
+            default:
+                out = "UNKNOWN(" + aeadAlg + ")";
+        }
+
+        switch (symAlg) {
+            case SymmetricKeyAlgorithmTags.AES_128:
+                out += " AES-128";
+                break;
+            case SymmetricKeyAlgorithmTags.AES_192:
+                out += " AES-192";
+                break;
+            case SymmetricKeyAlgorithmTags.AES_256:
+                out += " AES-256";
+                break;
+            default:
+                out += " UNKNOWN(" + symAlg + ")";
+        }
+        return out;
+    }
+
+    private void paddingPacketTests() throws IOException {
+        testRandom50Bytes();
+        testKnownPaddingBytes();
+        testPaddingPacketEncoding();
+    }
+
+    private void testKnownPaddingBytes() {
+        byte[] known = "thisIsKnownPadding".getBytes(Charset.forName("UTF8"));
+        PaddingPacket packet = new PaddingPacket(known);
+        isTrue(Arrays.areEqual(known, packet.getPadding()));
+    }
+
+    private void testRandom50Bytes() {
+        SecureRandom random = new SecureRandom();
+        PaddingPacket packet = new PaddingPacket(50, random);
+        isEquals(50, packet.getPadding().length);
+    }
+
+    private void testPaddingPacketEncoding() throws IOException {
+        SecureRandom random = new SecureRandom();
+        PaddingPacket packet = new PaddingPacket(123, random);
+
+        ByteArrayOutputStream bOut = new ByteArrayOutputStream();
+        BCPGOutputStream bcOut = new BCPGOutputStream(bOut, true);
```

Comment:
I noticed, that round-tripping the PaddingPacket only appears to work properly when `useNewFormat` is set to true. I may be misunderstanding the BCPG stream classes, but when I set `useNewFormat` to false, the PGPObjectFactory reads the next packet tag of the PaddingPacket as 20 instead of 21, which results in a parser error.

Reply:
Ah, glad that bug got uncovered and fixed so promptly! Thanks!